### PR TITLE
Expired listing inconsistency issue and additional settings

### PR DESCRIPTION
**Pull request by turtlepod**
There are several issues in how WPJM handle expired listing.
You can read it here: https://github.com/Automattic/WP-Job-Manager/issues/869

### This PR try to solve this by.
- Fix exp date consistency by setting listing status on post save.
- Do not change the exp listing date via JS (in meta box): also allow flexibility in setting expire date.
- Separate setting in handling expired listing in archive/job search vs single listing content

There also an interesting feature suggestion: Ability to redirect expired listing/set to 404:
https://github.com/Automattic/WP-Job-Manager/issues/823
I think it can easily added via add-on. I create an example plugin to do this:
https://github.com/turtlepod/wp-job-manager-redirect-expired-listing

Note: Please test before merge.

### CODE REVIEW

**Comment by dbtlr** [comment]
Can you please provide more detailed testing instructions?

**Comment by turtlepod** [comment]
1. First, switch to master branch
2. You will need to have an expired listing. (e.g listing with expired status, and expired date for example two weeks ago, etc)

eg-expired-listing

3. Now Edit it, and you will have this (it shows todays date, and not the actual expired date)

eg-expired-listing-edit


This will cause a lot of issue, user experience, data submit/save, etc.. (just messy)
Here's several issue that i think cause by this: #805, #822, #843, #860, #858, #865

I think you need to fix that.

There are other feature also added in this PR, but, I need you to check the above first and see if you can replicate the issue.

**Comment by turtlepod** [comment]
And try to update that expired listing too (try to keep the expired date as the correct expired date, not todays date)

so you'll understand what I mean by data submit/save issue and user experience issue.

File: includes/admin/class-wp-job-manager-writepanels.php [hunk 1]
```diff
@@ -85,8 +85,8 @@ public function job_listing_fields() {
 				'priority'    => 11,
 				'classes'     => array( 'job-manager-datepicker' ),
 				/* translators: date format placeholder, see https://secure.php.net/date */
-				'placeholder' => _x( 'yyyy-mm-dd', 'Date format placeholder.', 'wp-job-manager' ),
-				'value'       => metadata_exists( 'post', $post->ID, '_job_expires' ) ? get_post_meta( $post->ID, '_job_expires', true ) : calculate_job_expiry( $post->ID ),
+				'placeholder' => metadata_exists( 'post', $post->ID, '_job_expires' ) ? esc_attr_x( 'yyyy-mm-dd', 'Date format placeholder.', 'wp-job-manager' ) : calculate_job_expiry( $post->ID ),
```
Comment (jom) [inline includes/admin/class-wp-job-manager-writepanels.php]: No need to escape attribute (`_x()` is fine here). Attributes are escaped in templates.

File: wp-job-manager-functions.php [hunk 2]
```diff
@@ -23,7 +23,7 @@ function get_job_listings( $args = array() ) {
 		'fields'            => 'all'
 	) );
 
-	if ( false == get_option( 'job_manager_hide_expired_content', 1 ) ) {
+	if ( false == get_option( 'job_manager_hide_expired', get_option( 'job_manager_hide_expired_content' ) ) ) {
```
Comment (jom) [inline wp-job-manager-functions.php]: Do we want `get_option( 'job_manager_hide_expired_content' )` to have the original default value (`1`)?

File: includes/admin/class-wp-job-manager-writepanels.php [hunk 3]
```diff
@@ -556,6 +556,20 @@ public function save_job_listing_data( $post_id, $post ) {
 				}
 			}
 		}
+
+		/* Set Post Status To Expired If Already Expired */
+		$expiry_date = get_post_meta( $post_id, '_job_expires', true );
+		$today_date  = date( 'Y-m-d', current_time( 'timestamp' ) );
+		$post_status = $expiry_date && $today_date > $expiry_date ? 'expired' : false;
+		if( $post_status ){
```
Comment (jom) [inline includes/admin/class-wp-job-manager-writepanels.php]: Mind [adding space around parenthesis](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#space-usage) of this `if ( ... ) {}` statement?

**Comment by turtlepod** [comment]
I added the fix as requested 👍 nice catch!

**Comment by jom** [comment]
Tested. Looks good to me! Thanks, @turtlepod!

**Pull request merged**